message_ix_models/util/__init__.py

```diff
@@ -161,14 +161,29 @@ def aggregate_codes(df: pd.DataFrame, dim: str, codes):  # pragma: no cover
         print(key, group_series.replace({dim: mapping}))
 
 
-def broadcast(df, **kwargs):
+def broadcast(df, *args, **kwargs):
     """Fill missing data in `df` by broadcasting.
 
     Arguments
     ---------
+    args
+        tuble/dataframe mapping multiple dimensions to be fill.
     kwargs
         Keys are dimensions. Values are labels along that dimension to fill.
+ 
     """
+    labels = pd.DataFrame(*args)
+    if not labels.empty:
+        for cc in labels.columns:
+            if cc in df.columns:
```

Comment:
thanks! I am incorporating these changes.
Actually, I wanted the function to also work in case the column/dimensions in `labels` are not already present in `df` , i.e. cases where not necessarily `make_df()` is used.
following the example above, I wanted the function to work also for 

```python
inpN = inp0.drop(columns={'year_vtg','year_act'})
# this part of the function is not summorting this case
inp1 = inpN.pipe(broadcast, year_vtg=[2010,2020], 
           year_act=[2010,2020])
# this would work with the if condition
inp2 = inpN.pipe(broadcast, map_v_a)
```
Maybe it is too much, but it would make it more generic and multi-purposes

Reply:
Is this the same `inp0` from the PR description, above? If so, you could do this without drop():
```python
inpN = inp0.assign(year_vtg=None, year_act=None) 
inp1 = inpN.pipe(broadcast, …)
```

This produces the same result, but now the "strict" behaviour of broadcast() can help to detect errors, e.g. if there are some unintended extra dimensions in an argument like `map_v_a`.

Reply:
ok, agree